### DISCUSSION TITLE
Align ansis:LandSurveySite and ansis:SoilSite properties

### DISCUSSION THREAD

**Discussion by abhritchie** [ANZSoilData/def-au-schema-json#15] [General]
In the ANSIS Ontology `SoilSite` is a subclass of `LandSurveySite` so I'm assuming that properties and rules provided for the latter apply to the former (unless refined). I am however frightfully scared that that is UML thinking, not set thinking.

Most of the properties assigned (`schema:domainIncludes`) to a `LandSurveySite` have also been assigned to a `SoilSite`, except for:

- `aspect`
- `elevation`

I've added these to `SoilSite` in the JSON Schema.

Also, when cardinality is specified for a property on `LandSurveySite` I've reused that on `SoilSite` (e.g. for `hasLandCover`).

Correct decisions both?

**Comment by dr-shorthair** [comment]
> that is UML thinking, not set thinking 

How are they different? 
Set thinking says that every member of a subclass is also a member of the parent class (subset relationship), so it may also be described using the properties associated with the parent. 

And yes, that applies to cardinalities as well. 

Does JSON-Schema allow for sub-classing?

**Comment by abhritchie** [comment]
> How are they different?

I don't think they are, but limited experience with RDF/OWL and slightly confusion expression of the model in TTL/TopBraid gave me pause for thought.

**Reply by dr-shorthair** [reply]
Looking at Restrictions in TTL will give you headaches. 
Manchester syntax is easier to grok. 
(TopBraid form shows Manchester syntax).

**Reply by abhritchie** [reply]
Agree. I'm reading using TopBraid for all of this work ...
![image](https://user-images.githubusercontent.com/9596562/236099445-09345b2e-ef3c-4b55-b896-a6a5f7dcb089.png)

**Reply by dr-shorthair** [reply]
Just to complete the lesson: the way to understand these subClass relationships is 
- first line says 'every SoilSite is also a member of the class called LandsurveySite'
- next line says 'every SoilSite is also a member of the anonymous class of things with a maximum of 1 design-site properties with a value taken from the class SoilSite-design'
- next line says 'every SoilSite is also a member of the anonymous class of things with a maximum of 1 design-survey properties with a value taken from the class Survey-design'
- etc

Note that the qualified-cardinality constraint limits the count of design-site properties with that range, but says nothing about the count of design-site properties with other ranges. To absolutely limit the count of a property you need an unqualified cardinality constraint (e.g. `ansis:design-site max 1`). To absolutely constrain the range in this context, you should separately say `ansis:design-site only ansis:SoilSite-design` ... 

You have to keep a clear logical head, and be very aware of the OWA when using OWL restrictions. 

SHACL, OTOH, explicitly closes the world.

**Reply by abhritchie** [reply]
That's how I've been reading them. (Phew.)

The range dependent restriction is consistent with how I have to define properties in the JSON schema to ensure the correct enumeration (or data type) is used for a given class. For example, `abundance` has seven definitions for mottles, outcrops etc. Cardinality (strictly speaking, whether a property is listed as `required` and/or encoded as an object/literal or an array) is defined for the references to the specific definition.

To define an absolute constraint would be purely informative - I'm pretty sure validators don't have concept of restriction.

**Reply by abhritchie** [reply]
> SHACL, OTOH, explicitly closes the world.

I have wondered if we may be better off defining cardinality and some ranges using SHACL.

Easier for non-ontologists (nontologists) to understand and also separating invariant conceptual 'truth' from context/application specific restrictions.

**Reply by dr-shorthair** [reply]
Indeed. For data validation. You need SHACL. 
OWL is for reasoning.

**Comment by abhritchie** [comment]
> Does JSON-Schema allow for sub-classing?

Not as such. You can simulate it by creating and referencing discrete 'sub-schema' (e.g. individual schema for properties), but it breaks at a certain level of complexity. For example, when you need to refine properties a sub-class (say to the use of a specific vocabulary).

My approach is to reuse patterns across sub-classes.

**Reply by dr-shorthair** [reply]
Mixins. Or intersections of many classes.

**Comment by abhritchie** [comment]
BTW, if a property has no restriction defined on the class or super-class I have assumed a cardinality of none-to-many. I.e. it has no entry in the `required` property array and its values are `items` in an array.
